pkg/api/v1/atlasprivateendpoint_types.go

```diff
@@ -0,0 +1,157 @@
+	ProjectID string `json:"projectId"`
+	// GroupName is the label that identifies a set of endpoints.
+	// +kubebuilder:validation:Required
+	GroupName string `json:"groupName"`
+	// Endpoints is the list of individual private endpoints that comprise this endpoint group.
+	// +kubebuilder:validation:Required
+	Endpoints []GCPPrivateEndpoint `json:"endpoints"`
+}
+
+type GCPPrivateEndpoint struct {
+	// Name that identifies the Google Cloud consumer forwarding rule that you created.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// IP address to which this Google Cloud consumer forwarding rule resolves.
+	// +kubebuilder:validation:Required
+	IP string `json:"ipAddress"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +groupName:=atlas.mongodb.com
+// +kubebuilder:resource:categories=atlas,shortName=pe
+// +kubebuilder:printcolumn:name="Provider",type=string,JSONPath=`.spec.provider`
+// +kubebuilder:printcolumn:name="Region",type=string,JSONPath=`.spec.region`
+// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+
+// The AtlasPrivateEndpoint custom resource definition (CRD) defines a desired [Private Endpoint](https://www.mongodb.com/docs/atlas/security-private-endpoint/#std-label-private-endpoint-overview) configuration for an Atlas project.
+// It allows a private connection between your cloud provider and Atlas that doesn't send information through a public network.
+//
+// You can use private endpoints to create a unidirectional connection to Atlas clusters from your virtual network.
+type AtlasPrivateEndpoint struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AtlasPrivateEndpointSpec          `json:"spec,omitempty"`
+	Status status.AtlasPrivateEndpointStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// AtlasPrivateEndpointList contains a list of AtlasPrivateEndpoint
+type AtlasPrivateEndpointList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AtlasPrivateEndpoint `json:"items"`
+}
+
+func (pe *AtlasPrivateEndpoint) AtlasProjectObjectKey() client.ObjectKey {
+	ns := pe.Namespace
+	if pe.Spec.Project.Namespace != "" {
+		ns = pe.Spec.Project.Namespace
+	}
+
+	return kube.ObjectKey(ns, pe.Spec.Project.Name)
+}
+
+func (pe *AtlasPrivateEndpoint) Credentials() *api.LocalObjectReference {
+	return pe.Spec.Credentials()
+}
+
+func (pe *AtlasPrivateEndpoint) GetStatus() api.Status {
+	return pe.Status
+}
+
+func (pe *AtlasPrivateEndpoint) UpdateStatus(conditions []api.Condition, options ...api.Option) {
+	pe.Status.Conditions = conditions
+	pe.Status.ObservedGeneration = pe.ObjectMeta.Generation
+
+	for _, o := range options {
+		v := o.(status.AtlasPrivateEndpointStatusOption)
+		v(&pe.Status)
+	}
+}
+
+func init() {
```

Comment:
nit: please put `init()` functions on top of the file

Reply:
done